.github/workflows/pull_request.yml

```
@@ -0,0 +1,30 @@
+name: Pull Request Checks
+
+on:
+  pull_request:
+    branches: [ main ]
```

I'd suggest to run checks in every PR

```suggestion
    paths-ignore:
      - docs/**
      - public/**
      - *.md
      - *.yaml
      - .gitignore
```